File: dichotomy/dicho.js
function dicho(f, a, b, e) {
	const fx = (x) => eval(f);

	let n = 0;
	let x;

	while (Math.abs(b - a) > e) {
		n++;
		x = (a + b) / 2;

		if (fx(a) * fx(x) < 0) {
			b = x;
		} else {
			a = x;
		}
	}

	console.log(`solution: ${x}\nnumber of iterations: ${n}`);
}

File: dichotomy/dicho.go
package dicho

import (
	"math"
)

func Dicho(f func(float64) float64, a, b, e float64) (float64, int) {
	n := 0
	var x float64

	for math.Abs(b-a) > e {
		n++
		x = (a + b) / 2

		if f(a)*f(x) < 0 {
			b = x
		} else {
			a = x
		}
	}

	return x, n
}
